Game/Scripts/HeavyFinisherAttack.cpp

```diff
@@ -0,0 +1,315 @@
+
+	transform->SetLocalPosition(currentPos + vecForward * speed * deltaTime); //Move attack
+	if (rotateWhileAttacking)
+	{
+		transform->SetLocalRotation(Quat::RotateAxisAngle(float3::unitY, math::DegToRad(rotationVelocity * 10.0f * deltaTime))
+			* transform->GetLocalRotation());
+	}
+	transform->UpdateTransformMatrices();
+	rigidBody->UpdateRigidBody();
+
+	bool enemyHitten = false;
+	for (ComponentTransform* enemy : enemiesInTheArea)
+	{
+		if (currentPos.Distance(enemy->GetGlobalPosition()) < hitDistance)
+		{
+			target = enemy;
+			HitEnemy();
+			enemyHitten = true;
+			SeekNextEnemy();
+			attackHasTarget = true;
+			
+			break;
+		}
+	}
+
+	if (!enemyHitten)
+	{
+		if (currentPos.Distance(emptyAttackTargetPos) < hitDistance) //Attack gets to target
+		{
+			target = attackOwner;
+			isReturningToOwner = true;
+			attackHasTarget = true;
+			rigidBody->Disable();
+			if (!returnToPlayer)
+			{
+				ResetValues();
+			}
+		}
+	}
+}
+
+void HeavyFinisherAttack::HitEnemy()
+{
+	target->GetOwner()->GetComponent<HealthSystem>()->TakeDamage(damage);
+	audioSource->PostEvent(AUDIO::SFX::PLAYER::WEAPON::LIGHTSABER_CLASH);
+	vfx->SetPlayAtStart(true);
+	vfx->Play();
+	enemiesAlreadyHit.push_back(target);
+	countEnemyHits += 1.0f;
+}
+
+void HeavyFinisherAttack::PerformHeavyFinisher(ComponentTransform* target, ComponentTransform* attackOwner)
+{
+	this->target = target;
+	this->attackOwner = attackOwner;
+
+	attackHasTarget = true;
+
+	mesh->Enable();
+	rigidBody->Disable();
+
+	owner->SetParent(loadedScene->GetRoot());
+
+	isReturningToOwner = false;
+	isActivated = true;
+}
+
+void HeavyFinisherAttack::PerformEmptyHeavyFinisher(ComponentTransform* attackOwner)
+{
+	this->attackOwner = attackOwner;
+
+	emptyAttackTargetPos = attackOwner->GetGlobalPosition() + attackOwner->GetGlobalForward().Normalized() * defaultThrowDistance;
+
+	attackHasTarget = false;
+
+	mesh->Enable();
+	rigidBody->Enable();
+
+	owner->SetParent(loadedScene->GetRoot());
+
+	std::vector<ComponentRigidBody*> collisions;
+	physics->GetCollisions(rigidBody, collisions, "Enemy"); //Get enemies inside the collider
+
+	for (ComponentRigidBody* enemy : collisions)
+	{
+		enemiesInTheArea.push_back(enemy->GetOwner()->GetComponent<ComponentTransform>());
+	}
+
+	isReturningToOwner = false;
+	isActivated = true;
+}
+
+void HeavyFinisherAttack::SeekNextEnemy()
+{
+	target = nullptr;
+
+	rigidBody->Enable();
+	rigidBody->UpdateRigidBody();
+
+	std::vector<ComponentRigidBody*> collisions;
+	physics->GetCollisions(rigidBody, collisions, "Enemy"); //Get enemies inside the collider
+
+	float3 currentPos = transform->GetGlobalPosition();
+
+	for (ComponentRigidBody* enemy : collisions)
+	{
+		ComponentTransform* enemyTransform = enemy->GetOwner()->GetComponent<ComponentTransform>();
+
+		if (!enemy->GetOwner()->GetComponent<HealthSystem>()->EntityIsAlive() //If enemy is dead
+			|| (std::find(enemiesAlreadyHit.begin(), enemiesAlreadyHit.end(), enemyTransform)
```

Comment:
This works fine, but since you don't actually want to find that value, you just want to know if there's one, I'd suggest you use `std::any_of`. This higher order function returns true if at least one of the elements in the range matches your condition.
```suggestion
			|| (std::any_of(enemiesAlreadyHit.begin(), enemiesAlreadyHit.end(), [&enemyTransform](const ComponentTransform* other){ return other == enemyTransform; }))
```
(I'm on my phone, excuse the formatting :P)